scripts/update_dependencies.sh

```diff
@@ -0,0 +1,41 @@
+# Pip dependency update script
+# Usage:
+# bash <path to script> [Optional branch name]
+#
+# If the branch name is specified install the requirements from that branch
+# otherwise install from develop
+set -e
+
+if [ "$1" ]
+  then
+    REQ_BRANCH="$1"
+else
+    REQ_BRANCH="develop"
+fi
+
+
+# make sure bluemira is in the current environment
+BM_INSTALLED=$(pip list | grep bluemira | awk 'NR==1 {print $1}')
+[ "$BM_INSTALLED" = "bluemira" ] || { echo "bluemira not found in the current python environment, exiting" && exit 1; }
+
+readonly BLUEMIRA_ROOT="$(realpath "$(dirname "$0")"/..)"
+echo "Bluemira directory is $BLUEMIRA_ROOT"
+
+GIT_CMD="git -C ""$BLUEMIRA_ROOT"
+
+# check that the requirements files havent changed
+$GIT_CMD diff --exit-code requirements.txt \
+     || { echo "requirements.txt modified on this branch, exiting" && exit 1; }
+$GIT_CMD diff --exit-code requirements-develop.txt \
+     || { echo "requirements-develop.txt modified on this branch, exiting" && exit 1; }
+
+# checkout dependencies from develop
+$GIT_CMD checkout $REQ_BRANCH requirements.txt requirements-develop.txt
+
+# install dependencies
+pip install -r "$BLUEMIRA_ROOT/requirements.txt" || { echo "pip update failed" && exit 1; }
+pip install -r "$BLUEMIRA_ROOT/requirements-develop.txt" || { echo "pip update failed" && exit 1; }
```

Comment:
Shouldn't this install them with something like `python -m pip install --no-cache-dir -e .'[dev]'` or similar? or -U or --force-reinstall or whatever other pip flags are needed to properly update the environment? If only it were this simple! It's the pip flags and their many embodiments I struggle with on top of remembering to do this.

Reply:
All of the above flags are useful for some things like updating specific dependencies or if something isnt playing ball (eg because of some conda weirdness). That install method will install the latest (dev) version of bluemira in editable mode, not its dependencies, unless there is a change to the setup.py file. That is because we keep the setup.py file as loose as possible to not restrict developer of other codes and their needed dependency versions (with no guarantee of anything, but will probably work). 

If you install with the requirements files you get the same dependencies as our CI, it doesnt however touch the bluemira install itself which has no effect because we currently recommend editable mode